scanpipe/pipelines/get_dwarfs_from_elfs.py

```diff
@@ -0,0 +1,47 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# http://nexb.com and https://github.com/nexB/scancode.io
+# The ScanCode.io software is licensed under the Apache License version 2.0.
+# Data generated with ScanCode.io is provided as-is without warranties.
+# ScanCode is a trademark of nexB Inc.
+#
+# You may not use this software except in compliance with the License.
+# You may obtain a copy of the License at: http://apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+#
+# Data Generated with ScanCode.io is provided on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, either express or implied. No content created from
+# ScanCode.io should be considered or used as legal advice. Consult an Attorney
+# for any legal advice.
+#
+# ScanCode.io is a free software code scanning tool from nexB Inc. and others.
+# Visit https://github.com/nexB/scancode.io for support and download.
+
+from pathlib import Path
+
+from elf_inspector.dwarf import get_dwarf_paths
+
+from scanpipe.pipelines import Pipeline
+
+
+class GetDwarfsFromElfs(Pipeline):
+    """Get dwarfs from elfs."""
+
+    download_inputs = False
+    is_addon = True
+
+    @classmethod
+    def steps(cls):
+        return (cls.get_dwarfs_from_elfs,)
+
+    def get_dwarfs_from_elfs(self):
+        """
+        Update ``extra_data`` of project with
+        dwarf data extracted from elf files.
+        """
+        for elf in self.project.codebaseresources.elfs():
+            data = get_dwarf_paths(Path(self.project.codebase_path / elf.path))
+            self.project.update_extra_data({elf.path: data})
```

Comment:
Make sure to only trigger 1 db query rather than 1 per iteration.

Reply:
I have changed the query on a particular file rather than a project according to https://github.com/nexB/scancode.io/pull/1068#discussion_r1479601265